Specs/Renderer/ShaderProgramSpec.js

```diff
@@ -355,6 +358,27 @@ defineSuite([
         expect(renderFragment(context, sp)).toEqual([255, 255, 255, 255]);
     });
 
+    it('creates duplicate uniforms if precision of uniforms in vertex and fragment shader do not match', function() {
+        var highpSupported = ContextLimits.highpSupported;
+        ContextLimits._highpSupported = false;
+        var vs = 'attribute vec4 position; uniform float u_value; varying float v_value; void main() { gl_PointSize = 1.0; v_value = u_value + czm_viewport.z*0.2; gl_Position = position; }';
+        var fs = 'uniform float u_value; varying float v_value; void main() { gl_FragColor = vec4(u_value + v_value + czm_viewport.z*0.2); }';
+        sp = ShaderProgram.fromCache({
+            context : context,
+            vertexShaderSource : vs,
+            fragmentShaderSource : fs
+        });
+        var uniformMap = {
+            u_value : function() {
+                return 0.2;
+            }
+        };
+        expect(sp.allUniforms.u_value).toBeDefined();
+        expect(sp.allUniforms.u_value_f).toBeDefined();
+        expect(renderFragment(context, sp, uniformMap)).toEqual([204, 204, 204, 204]);
```

Comment:
This is most likely a fragile test, where it will pass on some systems, but not others.  Can we rework the shader to be more coarse-grained, e.g., black is failing, non-black is passing.